service/src/main/java/bio/terra/workspace/app/controller/ControlledAzureResourceApiController.java

```diff
@@ -380,6 +383,20 @@ public ResponseEntity<ApiCreatedControlledAzureBatchPool> createAzureBatchPool(
     AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
     var workspace =
         validateWorkspaceResourceCreationPermissions(userRequest, workspaceUuid, body.getCommon());
+    String userEmail = samService.getSamUser(userRequest).getEmail();
```

Comment:
The logic here should likely be in a stairway step, instead of the API controller. For reference, [this](https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/AssignManagedIdentityAzureVmStep.java) is the step which requests a Sam pet and assigns it to Azure VMs.